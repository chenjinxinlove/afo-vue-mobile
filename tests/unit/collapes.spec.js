import {mount} from '@vue/test-utils'
import Demo from '../../src/collapse/demo'
import Vue from 'vue'
import Collapse from '../../src/collapse'
import CollapseItem from '../../src/collapse-item'

describe('Collapse.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Collapse)
    expect(Vue.component(Collapse.name))
      .toHaveProperty('name', 'VueComponent')
  })
  it('use', () => {
    Vue.use(CollapseItem)
    expect(Vue.component(CollapseItem.name))
      .toHaveProperty('name', 'VueComponent')
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(mount(Demo)).toMatchSnapshot()
  })
})
