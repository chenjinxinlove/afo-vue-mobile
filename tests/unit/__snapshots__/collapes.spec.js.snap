// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse.vue demo snapshot 1`] = `
<div>
  <p>基础用法</p>
  <div class="afo-collapse afo-hairline--top-bottom">
    <div class="afo-collapse-item afo-collapse-item--expanded">
      <div class="afo-cell afo-cell--clickable afo-hairlines afo-collapse-item__title">
        <div class="afo-cell__title"><span>1</span></div>
        <i class="afo-icon afo-icon-arrow afo-cell__right-icon"></i>
      </div>
      <div class="afo-collapse-item__wrapper">
        <div class="afo-collapse-item__content">
          111111111
        </div>
      </div>
    </div>
    <div class="afo-collapse-item afo-hairline--top">
      <div class="afo-cell afo-cell--clickable afo-hairlines afo-collapse-item__title">
        <div class="afo-cell__title"><span>2</span></div>
        <i class="afo-icon afo-icon-arrow afo-cell__right-icon"></i>
      </div>
      <!---->
    </div>
    <div class="afo-collapse-item afo-hairline--top">
      <div class="afo-cell afo-cell--clickable afo-hairlines afo-collapse-item__title">
        <div class="afo-cell__title"><span>3</span></div>
        <i class="afo-icon afo-icon-arrow afo-cell__right-icon"></i>
      </div>
      <!---->
    </div>
  </div>
  <p>手风琴</p>
  <div class="afo-collapse afo-hairline--top-bottom">
    <div class="afo-collapse-item">
      <div class="afo-cell afo-cell--clickable afo-hairlines afo-collapse-item__title">
        <div class="afo-cell__title"><span>1</span></div>
        <i class="afo-icon afo-icon-arrow afo-cell__right-icon"></i>
      </div>
      <!---->
    </div>
    <div class="afo-collapse-item afo-hairline--top">
      <div class="afo-cell afo-cell--clickable afo-hairlines afo-collapse-item__title">
        <div class="afo-cell__title"><span>2</span></div>
        <i class="afo-icon afo-icon-arrow afo-cell__right-icon"></i>
      </div>
      <!---->
    </div>
    <div class="afo-collapse-item afo-hairline--top">
      <div class="afo-cell afo-cell--clickable afo-hairlines afo-collapse-item__title">
        <div class="afo-cell__title"><span>3</span></div>
        <i class="afo-icon afo-icon-arrow afo-cell__right-icon"></i>
      </div>
      <!---->
    </div>
  </div>
  <p>自定义标题内容</p>
  <div class="afo-collapse afo-hairline--top-bottom">
    <div class="afo-collapse-item afo-collapse-item--expanded">
      <div class="afo-cell afo-cell--clickable afo-hairlines afo-collapse-item__title">
        <div class="afo-cell__title">
          <div>1
            <afo-icon type="question" size="16px"></afo-icon>
          </div>
        </div>
        <i class="afo-icon afo-icon-arrow afo-cell__right-icon"></i>
      </div>
      <div class="afo-collapse-item__wrapper">
        <div class="afo-collapse-item__content">
          11111111111
        </div>
      </div>
    </div>
    <div class="afo-collapse-item afo-hairline--top">
      <div class="afo-cell afo-cell--clickable afo-hairlines afo-collapse-item__title">
        <div class="afo-cell__title"><span>2</span></div>
        <i class="afo-icon afo-icon-arrow afo-cell__right-icon"></i>
      </div>
      <!---->
    </div>
  </div>
</div>
`;
